cmd/in/main.go

```diff
@@ -92,54 +72,59 @@ func main() {
 		return
 	}
 
-	fmt.Fprintf(os.Stderr, "fetching %s@%s\n", color.GreenString(req.Source.Repository), color.YellowString(req.Version.Digest))
+	tag := repo.Tag(req.Version.Tag)
 
-	var image v1.Image
-	digest := new(name.Digest)
+	if !req.Params.SkipDownload {
```

Comment:
Hey, I might have missed the purpose of refactoring in/main.go, moving the request and response struct definitions looks good. But it appears as though the the main block has one extra level of nesting to be able to move the response writing section to the bottom.

IMHO, might be simpler and more readable to extract the response writing to a reusable private function and keep the main block flatter. WDYT ?

Reply:
Ah right - sorry, at one point I was trying to fix #227 but it looks like I committed some partial work here.

If you don't mind the additional content to review, I think I'll carry forward with that so I can refactor the `RegistryMirror` usage in `in` to match the refactor I made to `check` at the same time.